mount/temp.go

```diff
@@ -67,6 +68,27 @@ func WithTempMount(ctx context.Context, mounts []Mount, f func(root string) erro
 	return nil
 }
 
+// removeVolatileTempMount The volatile option of overlayfs doesn't allow to mount again using the
+// same upper / work dirs. Since it's a temp mount, avoid using that
+// option here. Reference: https://docs.kernel.org/filesystems/overlayfs.html#volatile-mount
+// TODO: Make this logic conditional once the kernel supports reusing
+// overlayfs volatile mounts.
+func removeVolatileTempMount(mounts []Mount) []Mount {
+	for _, m := range mounts {
+		if m.Type != "overlay" {
+			continue
+		}
+		for i, opt := range m.Options {
+			if opt == "volatile" {
+				m.Options[i] = ""
```

Comment:
This is changing the input slice, if the "volatile" option is detected, the original slice and options should be copied to avoid the mount call input slice.

Reply:
@dmcgowan I am not sure I follow. The original slice are still copied over. My test here should cover that `TestRemoveVolatileTempMount`. Maybe I'm missing something?

Reply:
I should have posted the test update with my comment, sorry. The input slice is never copied.

```.diff

diff --git a/mount/mount_test.go b/mount/mount_test.go
index 9fc7e0165..078c6c300 100644
--- a/mount/mount_test.go
+++ b/mount/mount_test.go
@@ -205,9 +205,24 @@ func TestRemoveVolatileTempMount(t *testing.T) {
        }

        for _, tc := range testCases {
+               original := copyMounts(tc.input)
                actual := removeVolatileTempMount(tc.input)
                if !reflect.DeepEqual(actual, tc.expected) {
                        t.Fatalf("incorrectly modified mounts: %s.\n\n Expected: %v\n\n, Actual: %v", tc.desc, tc.expected, actual)
                }
+               if !reflect.DeepEqual(original, tc.input) {
+                       t.Fatalf("modified original mounts: %s.\n\n Expected: %v\n\n, Actual: %v", tc.desc, original, tc.input)
+               }
+       }
+}
+
+func copyMounts(in []Mount) []Mount {
+       out := make([]Mount, len(in))
+       for i := range in {
+               out[i] = in[i]
+               if len(out[i].Options) > 0 {
+                       out[i].Options = append([]string{out[i].Options[0]}, out[i].Options[1:]...)
+               }
        }
+       return out
 }
```

Reply:
ahh that makes sense! I've adjusted the code to not modify the original slice! Please take another look. And it looks like I need to rebase so I will do that too :)